lntest/itest/lnd_network_test.go

```diff
@@ -118,98 +117,71 @@ func testReconnectAfterIPChange(net *lntest.NetworkHarness, t *harnessTest) {
 		daveArgs = append(daveArgs, "--externalip="+addr)
 	}
 
-	// withP2PPort is a helper closure used to set the P2P port that a node
-	// should use.
-	var withP2PPort = func(port int) lntest.NodeOption {
-		return func(cfg *lntest.BaseNodeConfig) {
-			cfg.P2PPort = port
-		}
-	}
-
 	// Create a new node, Dave, and ensure that his initial P2P port is
 	// ip1 derived above.
-	dave := net.NewNode(t.t, "Dave", daveArgs, withP2PPort(ip1))
-	defer shutdownAndAssert(net, t, dave)
-
-	// Subscribe to graph notifications from Charlie so that we can tell
-	// when he receives Dave's NodeAnnouncements.
-	ctxb := context.Background()
-	charlieSub := subscribeGraphNotifications(ctxb, t, charlie)
-	defer close(charlieSub.quit)
+	dave := ht.NewNode("Dave", daveArgs)
```

Comment:
I wonder why this test ever worked, given we tried to connect to a port that Dave never listened on?

Reply:
Changed the test a bit to make it more explicit. When we create a new node, we'd auto-assign a p2p port.